tor_core/config.py

```diff
@@ -11,6 +13,51 @@
 from tor_core import __version__
 
 
+_missing = object()
+
+
+# @see https://stackoverflow.com/a/17487613/1236035
+class cached_property(object):
+    """A decorator that converts a function into a lazy property.  The
+    function wrapped is called the first time to retrieve the result
+    and then that calculated result is used the next time you access
+    the value::
+
+        class Foo(object):
+
+            @cached_property
+            def foo(self):
+                # calculate something important here
+                return 42
+
+    The class has to have a `__dict__` in order for this property to
+    work.
+    """
+
+    # implementation detail: this property is implemented as non-data
+    # descriptor.  non-data descriptors are only invoked if there is
```

Comment:
Can I totally nitpick and ask to remove the double spaces at the beginning of the sentences here?

Reply:
Totally fair. This was copy-pasted from the stack overflow answer linked above so I left the original commenting here too.